error convert `xsd:list` members back to string

#48 fixes parsing `xsd:list` in model generation (#35), but exposed a new problem when writing XML:

```
xmlschema.validators.exceptions.XMLSchemaEncodeError:
failed validating <Type.TIME_LAPSE: 'TimeLapse'> with XsdAtomicBuiltin(name='xs:string')

Reason: <Type.TIME_LAPSE: 'TimeLapse'> is not an instance of <class 'str'>.
```

I'm not certain yet whether to failure to convert these Enums back to `str` has to do with coming from a `xsd:list`?

<details>

```pytb
_________________________________________________________________________ test_roundtrip[tubhiswt] __________________________________________________________________________

xml = '/Users/talley/Dropbox (HMS)/Python/ome-types/testing/data/tubhiswt.ome.xml'

    @pytest.mark.parametrize("xml", xml_roundtrip, ids=true_stem)
    def test_roundtrip(xml):
        """Ensure we can losslessly round-trip XML through the model and back."""
        xml = str(xml)
        schema = get_schema(xml)

        def canonicalize(xml, strip_empty):
            d = schema.decode(xml, use_defaults=True)
            # Strip extra whitespace in the schemaLocation value.
            d["@xsi:schemaLocation"] = re.sub(r"\s+", " ", d["@xsi:schemaLocation"])
            root = schema.encode(d, path=NS_OME + "OME", use_defaults=True)
            # These are the tags that appear in the example files with empty
            # content. Since our round-trip will drop empty elements, we'll need to
            # strip them from the "original" documents before comparison.
            if strip_empty:
                for tag in ("Description", "LightPath", "Map"):
                    for e in root.findall(f".//{NS_OME}{tag}[.='']..."):
                        e.remove(e.find(f"{NS_OME}{tag}"))
            # ET.canonicalize can't handle an empty namespace so we need to
            # re-register the OME namespace with an actual name before calling
            # tostring.
            ElementTree.register_namespace("ome", URI_OME)
            xml_out = ElementTree.tostring(root, "unicode")
            xml_out = util.canonicalize(xml_out, strip_text=True)
            xml_out = minidom.parseString(xml_out).toprettyxml(indent="  ")
            return xml_out

        original = canonicalize(xml, True)
>       ours = canonicalize(to_xml(from_xml(xml)), False)

testing/test_model.py:111:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/ome_types/schema.py:283: in to_xml
    root = to_xml_element(ome)
src/ome_types/schema.py:269: in to_xml_element
    root = schema.encode(
../../../miniconda3/envs/ome/lib/python3.8/site-packages/xmlschema/validators/schema.py:1638: in encode
    for result in self.iter_encode(obj, path, validation, *args, **kwargs):
../../../miniconda3/envs/ome/lib/python3.8/site-packages/xmlschema/validators/schema.py:1625: in iter_encode
    yield from xsd_element.iter_encode(obj, validation, converter=converter,
../../../miniconda3/envs/ome/lib/python3.8/site-packages/xmlschema/validators/elements.py:864: in iter_encode
    for result in xsd_type.content.iter_encode(element_data, validation, **kwargs):
../../../miniconda3/envs/ome/lib/python3.8/site-packages/xmlschema/validators/groups.py:794: in iter_encode
    for result in xsd_element.iter_encode(value, validation, **kwargs):
../../../miniconda3/envs/ome/lib/python3.8/site-packages/xmlschema/validators/elements.py:874: in iter_encode
    yield self.validation_error(validation, e, elem, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = XsdElement(ref='OME:Experiment', occurs=[0, None]), validation = 'strict'
error = XMLSchemaEncodeError(reason="<Type.TIME_LAPSE: 'TimeLapse'> is not an instance of <class 'str'>.")
obj = <Element '{http://www.openmicroscopy.org/Schemas/OME/2016-06}Experiment' at 0x113a7def0>, source = None, namespaces = None
_kwargs = {'converter': <ome_types.schema.OMEConverter object at 0x113bef940>, 'level': 1, 'unordered': False, 'use_defaults': False}

    def validation_error(self, validation, error, obj=None,
                         source=None, namespaces=None, **_kwargs):
        """
        Helper method for generating and updating validation errors. If validation
        mode is 'lax' or 'skip' returns the error, otherwise raises the error.

        :param validation: an error-compatible validation mode: can be 'lax' or 'strict'.
        :param error: an error instance or the detailed reason of failed validation.
        :param obj: the instance related to the error.
        :param source: the XML resource related to the validation process.
        :param namespaces: is an optional mapping from namespace prefix to URI.
        :param _kwargs: keyword arguments of the validation process that are not used.
        """
        check_validation_mode(validation)
        if isinstance(error, XMLSchemaValidationError):
            if error.namespaces is None and namespaces is not None:
                error.namespaces = namespaces
            if error.source is None and source is not None:
                error.source = source
            if error.obj is None and obj is not None:
                error.obj = obj
            if error.elem is None and is_etree_element(obj):
                error.elem = obj
        elif isinstance(error, Exception):
            error = XMLSchemaValidationError(self, obj, str(error), source, namespaces)
        else:
            error = XMLSchemaValidationError(self, obj, error, source, namespaces)

        if validation == 'strict' and error.elem is not None:
>           raise error
E           xmlschema.validators.exceptions.XMLSchemaEncodeError: failed validating <Type.TIME_LAPSE: 'TimeLapse'> with XsdAtomicBuiltin(name='xs:string'):
E
E           Reason: <Type.TIME_LAPSE: 'TimeLapse'> is not an instance of <class 'str'>.
E
E           Schema:
E
E             <xs:simpleType xmlns:hfp="http://www.w3.org/2001/XMLSchema-hasFacetAndProperty" xmlns:xs="http://www.w3.org/2001/XMLSchema" name="string" id="string">
E               <xs:annotation>
E                 <xs:appinfo>
E                   <hfp:hasFacet name="length" />
E                   <hfp:hasFacet name="minLength" />
E                   <hfp:hasFacet name="maxLength" />
E                   <hfp:hasFacet name="pattern" />
E                   <hfp:hasFacet name="enumeration" />
E                   <hfp:hasFacet name="whiteSpace" />
E                   <hfp:hasProperty name="ordered" value="false" />
E                   <hfp:hasProperty name="bounded" value="false" />
E                   <hfp:hasProperty name="cardinality" value="countably infinite" />
E                   <hfp:hasProperty name="numeric" value="false" />
E                 </xs:appinfo>
E                 <xs:documentation source="http://www.w3.org/TR/xmlschema-2/#string" />
E               </xs:annotation>
E               <xs:restriction base="xs:anySimpleType">
E                 <xs:whiteSpace value="preserve" id="string.preserve" />
E               </xs:restriction>
E             </xs:simpleType>
E
E           Instance:
E
E             <ome:Experiment xmlns:ome="http://www.openmicroscopy.org/Schemas/OME/2016-06" ID="urn:lsid:loci.wisc.edu:Experiment:OWS350" Type="">
E                 <ome:Description>4 Cell Embryo</ome:Description>
E                 <ome:ExperimenterRef ID="urn:lsid:loci.wisc.edu:Experimenter:116" />
E             </ome:Experiment>

../../../miniconda3/envs/ome/lib/python3.8/site-packages/xmlschema/validators/xsdbase.py:906: XMLSchemaEncodeError
========================================================================== short test summary info ==========================================================================
FAILED testing/test_model.py::test_roundtrip[tubhiswt] - xmlschema.validators.exceptions.XMLSchemaEncodeError: failed validating <Type.TIME_LAPSE: 'TimeLapse'> with XsdAt...
```

</details>

--- a/src/ome_autogen.py
+++ b/src/ome_autogen.py
@@ -439,6 +439,8 @@ def make_dataclass(component: Union[XsdComponent, XsdType]) -> List[str]:
 def make_enum(component: XsdComponent) -> List[str]:
     name = component.local_name
     _type = component.type if hasattr(component, "type") else component
+    if _type.is_list():
+        _type = _type.item_type
     lines = ["from enum import Enum", ""]
     lines += [f"class {name}(Enum):"]
     doc = get_docstring(component, summary=True)
@@ -499,6 +501,11 @@ def iter_members(
         yield from iter_all_members(component)
 
 
+def is_enum_type(obj: XsdType) -> bool:
+    """Return true if XsdType represents an enumeration."""
+    return obj.get_facet(qnames.XSD_ENUMERATION) is not None
+
+
 class Member:
     def __init__(self, component: Union[XsdElement, XsdAttribute]):
         self.component = component
@@ -543,10 +550,6 @@ class Member:
         desc = get_docstring(self.component)
         desc = re.sub(r"\s?\[.+\]", "", desc)  # remove bracketed types
         return Parameter(self.identifier, _type, wrap(desc))
-
-    @property
-    def is_enum_type(self) -> bool:
-        return self.type.get_facet(qnames.XSD_ENUMERATION) is not None
 
     @property
     def is_builtin_type(self) -> bool:
@@ -609,7 +612,9 @@ class Member:
         # FIXME: this bit is mostly hacks
         if self.type.is_complex() and self.component.ref is None:
             locals_.append("\n".join(make_dataclass(self.component)) + "\n")
-        if self.type.is_restriction() and self.is_enum_type:
+        if self.type.is_restriction() and is_enum_type(self.type):
+            locals_.append("\n".join(make_enum(self.component)) + "\n")
+        if self.type.is_list() and is_enum_type(self.type.item_type):
             locals_.append("\n".join(make_enum(self.component)) + "\n")
         return locals_
 
@@ -667,7 +672,7 @@ class Member:
 
         if self.type.is_global():
             return self.type.local_name
-        elif self.type.is_complex():
+        elif self.type.is_complex() or self.type.is_list():
             return self.component.local_name
 
         if self.type.is_restriction():
@@ -708,7 +713,7 @@ class Member:
         else:
             default_val = self.component.default
             if default_val is not None:
-                if self.is_enum_type:
+                if is_enum_type(self.type):
                     default_val = f"{self.type_string}('{default_val}')"
                 elif hasattr(builtins, self.type_string):
                     default_val = repr(getattr(builtins, self.type_string)(default_val))
@@ -720,7 +725,8 @@ class Member:
 
     @property
     def max_occurs(self) -> bool:
-        return getattr(self.component, "max_occurs", 1)
+        default = None if self.type.is_list() else 1
+        return getattr(self.component, "max_occurs", default)
 
     @property
     def is_optional(self) -> bool:

--- a/src/ome_types/schema.py
+++ b/src/ome_types/schema.py
@@ -221,7 +221,9 @@ class OMEConverter(XMLSchemaConverter):
             name = _plural_to_singular.get(name, name)
             name = _snake_to_camel.get(name, name)
             if name in xsd_element.attributes:
-                if isinstance(value, Enum):
+                if isinstance(value, list):
+                    value = [getattr(i, "value", i) for i in value]
+                elif isinstance(value, Enum):
                     value = value.value
                 elif isinstance(value, datetime):
                     value = DateTime10.fromdatetime(value)

--- a/testing/data/tubhiswt.ome.xml
+++ b/testing/data/tubhiswt.ome.xml
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?><!-- Warning: this comment is an OME-XML metadata block, which contains crucial dimensional parameters and other important metadata. Please edit cautiously (if at all), and back up the original data before doing so. For more information, see the OME-TIFF web site: http://www.openmicroscopy.org/site/support/ome-model/ome-tiff/. -->
 <OME xmlns="http://www.openmicroscopy.org/Schemas/OME/2016-06"
     xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Creator="OME Bio-Formats 5.2.0-m4" UUID="urn:uuid:06846f98-85ab-43f1-b665-2ba8b11ca845" xsi:schemaLocation="http://www.openmicroscopy.org/Schemas/OME/2016-06 http://www.openmicroscopy.org/Schemas/OME/2016-06/ome.xsd">
-    <Experiment ID="urn:lsid:loci.wisc.edu:Experiment:OWS350" Type="TimeLapse">
+    <Experiment ID="urn:lsid:loci.wisc.edu:Experiment:OWS350" Type="TimeLapse SpectralImaging">
         <Description>4 Cell Embryo</Description>
         <ExperimenterRef ID="urn:lsid:loci.wisc.edu:Experimenter:116"/>
     </Experiment>

--- a/testing/test_model.py
+++ b/testing/test_model.py
@@ -25,7 +25,6 @@ SHOULD_FAIL_ROUNDTRIP = {
     "transformations-downgrade",
     "transformations-upgrade",
     # type Enum not being converted to str
-    "tubhiswt",
 }
 SKIP_ROUNDTRIP = {
     # These have XMLAnnotations with extra namespaces and mixed content, which

--- a/testing/test_model.py
+++ b/testing/test_model.py
@@ -17,7 +17,6 @@ import util  # isort: skip
 SHOULD_FAIL_READ = {
     # Some timestamps have negative years which datetime doesn't support.
     "timestampannotation",
-    "tubhiswt",
 }
 SHOULD_RAISE_READ = {"bad"}
 SHOULD_FAIL_ROUNDTRIP = {
@@ -25,6 +24,8 @@ SHOULD_FAIL_ROUNDTRIP = {
     "timestampannotation-posix-only",
     "transformations-downgrade",
     "transformations-upgrade",
+    # type Enum not being converted to str
+    "tubhiswt",
 }
 SKIP_ROUNDTRIP = {
     # These have XMLAnnotations with extra namespaces and mixed content, which

--- a/testing/test_model.py
+++ b/testing/test_model.py
@@ -24,7 +24,6 @@ SHOULD_FAIL_ROUNDTRIP = {
     "timestampannotation-posix-only",
     "transformations-downgrade",
     "transformations-upgrade",
-    # type Enum not being converted to str
 }
 SKIP_ROUNDTRIP = {
     # These have XMLAnnotations with extra namespaces and mixed content, which

